### PR TITLE
React list of goods

### PR DESCRIPTION
**Pull request by vsupruniuk**
[DEMO LINK](https://vsupruniuk.github.io/react_list-of-goods/)

### CODE REVIEW

File: src/App.tsx [hunk 1]
```diff
@@ -14,11 +16,208 @@ const goodsFromServer: string[] = [
...
+                <option value="1">1</option>
+                <option value="2">2</option>
+                <option value="3">3</option>
+                <option value="4">4</option>
+                <option value="5">5</option>
+                <option value="6">6</option>
+                <option value="7">7</option>
+                <option value="8">8</option>
+                <option value="9">9</option>
+                <option value="10">10</option>
```
Comment (stanislav-topikha) [inline src/App.tsx:213]: maybe loop can handle it?)

File: src/App.tsx [hunk 2]
```diff
@@ -14,11 +16,208 @@ const goodsFromServer: string[] = [
...
+
+  reverse = () => {
+    this.setState(state => ({
+      isReversed: !state.isReversed,
+    }));
+  };
+
+  setMinLength = (value: number) => {
+    this.setState({ minLength: value });
+  };
```
Comment (stanislav-topikha) [inline src/App.tsx:67]: use correct naming for event handlers: 
https://javascript.plainenglish.io/handy-naming-conventions-for-event-handler-functions-props-in-react-fc1cbb791364

**Review by stanislav-topikha — CHANGES_REQUESTED** [review]
Overall good 👍🏻 

but read the comments below ⏬ 

and consider making this button's separate, not in a group
image

**Review by stanislav-topikha — APPROVED** [review]

File: src/App.tsx [hunk 3]
```diff
@@ -14,11 +16,206 @@ const goodsFromServer: string[] = [
...
+          break;
+
+        default:
+          return 0;
+      }
+    }
+
+    if (isReversed) {
+      visibleGoods.reverse();
+    }
```
Comment (Olexandr6) [inline src/App.tsx:113]: This logic shouldn't be here. Method render should be as clear as possible.
Let's move this logic to a separate method

**Review by Olexandr6 — CHANGES_REQUESTED** [review]

**Review by Olexandr6 — APPROVED** [review]

**Pull request closed without merging**